client/client.go

```diff
@@ -181,49 +180,17 @@ func GetEntryBundle(ctx context.Context, f EntryBundleFetcherFunc, i, logSize ui
 // Since the tiles commit only to immutable nodes, the job of building proofs is slightly
 // more complex as proofs can touch "ephemeral" nodes, so these need to be synthesized.
 type ProofBuilder struct {
-	cp        log.Checkpoint
+	treeSize  uint64
 	nodeCache nodeCache
 }
 
 // NewProofBuilder creates a new ProofBuilder object for a given tree size.
 // The returned ProofBuilder can be re-used for proofs related to a given tree size, but
 // it is not thread-safe and should not be accessed concurrently.
-func NewProofBuilder(ctx context.Context, cp log.Checkpoint, f TileFetcherFunc) (*ProofBuilder, error) {
-	ctx, span := tracer.Start(ctx, "tessera.client.NewProofBuilder")
-	defer span.End()
-
-	span.SetAttributes(logSizeKey.Int64(otel.Clamp64(cp.Size)))
-
+func NewProofBuilder(ctx context.Context, treeSize uint64, f TileFetcherFunc) (*ProofBuilder, error) {
```

Comment:
Do we need to mention "no correctness checking of the root hash for the given tree size is performed." in this function doc comment?

Reply:
I don't think so, it's probably more surprising that a call to this function would go off and start making network requests before calling one of the `.*Proof` functions below.